packages/core/data-transfer/src/engine/index.ts

```diff
@@ -591,27 +633,99 @@ class TransferEngine<
     }
   }
 
+  #schemaDiffs: Record<string, Diff[]> = {};
+
   async integrityCheck() {
-    try {
-      const sourceMetadata = await this.sourceProvider.getMetadata();
-      const destinationMetadata = await this.destinationProvider.getMetadata();
-
-      if (sourceMetadata && destinationMetadata) {
-        this.#assertStrapiVersionIntegrity(
-          sourceMetadata?.strapi?.version,
-          destinationMetadata?.strapi?.version
-        );
-      }
+    const sourceMetadata = await this.sourceProvider.getMetadata();
+    const destinationMetadata = await this.destinationProvider.getMetadata();
+
+    if (sourceMetadata && destinationMetadata) {
+      this.#assertStrapiVersionIntegrity(
+        sourceMetadata?.strapi?.version,
+        destinationMetadata?.strapi?.version
+      );
+    }
 
-      const sourceSchemas = (await this.sourceProvider.getSchemas?.()) as SchemaMap;
-      const destinationSchemas = (await this.destinationProvider.getSchemas?.()) as SchemaMap;
+    const sourceSchemas = (await this.sourceProvider.getSchemas?.()) as SchemaMap;
+    const destinationSchemas = (await this.destinationProvider.getSchemas?.()) as SchemaMap;
 
+    try {
       if (sourceSchemas && destinationSchemas) {
         this.#assertSchemasMatching(sourceSchemas, destinationSchemas);
       }
     } catch (error) {
-      if (error instanceof Error) {
-        this.#panic(error);
+      // if this is a schema matching error
+      if (error instanceof TransferEngineValidationError && error.details?.details?.diffs) {
+        this.#schemaDiffs = error.details?.details?.diffs as Record<string, Diff[]>;
+
+        const context = {
+          diffs: this.#schemaDiffs,
+          source: this.sourceProvider,
+          destination: this.destinationProvider,
+        };
+
+        let workflowsStatus;
+        const source = 'Schema Integrity';
+
+        Object.entries(context.diffs).forEach(([uid, diffs]) => {
+          for (const diff of diffs) {
+            const path = `${uid}.${diff.path.join('.')}`;
+            const endPath = diff.path[diff.path.length - 1];
+
+            // Catch known features
+            // TODO: can this be moved outside of the engine?
```

Comment:
This is not for things that aren't transferrable, this is just to make error messaging nicer.  It collects anything that comes from review workflows and instead of printing them each as a schema difference, lists it all as one line called "review workflows"

It might work to move all of the logging out of this method and have it be the responsibility of the diff handlers. I'll see what I can do :)